web/src/buddy_list.js

```diff
@@ -55,16 +87,210 @@ class BuddyListConf {
+                        tooltip_text = $t(
+                            {
+                                defaultMessage:
+                                    "{N, plural, one {# other user} other {# other users}}",
+                            },
+                            {N: other_users_count},
+                        );
+                    }
+                    instance.setContent(tooltip_text);
+                },
+                onHidden(instance) {
+                    instance.destroy();
+                },
+                appendTo: () => document.body,
+            });
+        });
+        this.tooltips_initialized = true;
+    }
 
     populate(opts) {
         this.render_count = 0;
-        this.$container.empty();
+        this.$users_matching_view_container.empty();
+        this.users_matching_view_ids = [];
+        this.$other_users_container.empty();
+        this.other_user_ids = [];
 
         // We rely on our caller to give us items
         // in already-sorted order.
         this.all_user_ids = opts.all_user_ids;
 
         this.fill_screen_with_content();
+        if (!this.tooltips_initialized) {
+            this.initialize_tooltips();
+        }
+
+        // We do a handful of things once we're done rendering all the users,
+        // and each of these tasks need shared data that we'll compute first.
+        const current_sub = narrow_state.stream_sub();
+        const pm_ids_set = narrow_state.pm_ids_set();
+
+        // If we have only "other users" and aren't in a stream/DM view
+        // then we don't show section headers and only show one untitled
+        // section.
+        const hide_headers = this.should_hide_headers(current_sub, pm_ids_set);
+        const subscriber_count = this.total_subscriber_count(current_sub, pm_ids_set);
+        const total_user_count = people.get_active_human_count();
+        const other_users_count = total_user_count - subscriber_count;
+        const has_inactive_users_matching_view =
+            subscriber_count > this.users_matching_view_ids.length;
+        const has_inactive_other_users = other_users_count > this.other_user_ids.length;
+
+        const data = {
+            current_sub,
+            hide_headers,
+            subscriber_count,
+            other_users_count,
+            has_inactive_users_matching_view,
+            has_inactive_other_users,
+        };
+
+        $("#buddy-list-users-matching-view-container .view-all-subscribers-link").remove();
+        $("#buddy-list-other-users-container .view-all-users-link").remove();
+        if (!buddy_data.get_is_searching_users()) {
+            this.render_view_user_list_links(data);
+        }
+        this.render_section_headers(data);
+        if (!hide_headers) {
+            this.update_empty_list_placeholders(data);
+        }
+    }
+
+    update_empty_list_placeholders({has_inactive_users_matching_narrow, has_inactive_other_users}) {
```

Comment:
`has_inactive_users_matching_narrow` should be `has_inactive_users_matching_view`